blurring_as_a_service/cleaning_pipeline/source/sampling_pipeline.py

```diff
@@ -0,0 +1,458 @@
+        self.customer_quality_check_folder = customer_quality_check_folder
+        self.customer_retraining_folder = customer_retraining_folder
+        self.database_parameters_json = database_parameters_json
+        self.customer_name = customer_name
+        self.sampling_parameters = sampling_parameters
+        
+    def sample_images_for_quality_check(self, grouped_images_by_date: Dict[str, List[str]] 
+    ) -> None:
+        """
+        Samples a specified number of images from each date for quality checking.
+
+        Parameters
+        ----------
+        grouped_images_by_date : Dict[str, List[str]]
+            A dictionary where keys are dates and values are lists of image file names from those dates.
+        input_structured_folder : str
+            The path of the input folder containing images.
+        customer_quality_check_folder : str
+            The destination folder path for the quality check images.
+        n_images_to_sample : int
+            Number of images to sample for each date.
+            
+        Returns
+        -------
+        None
+        """
+        
+        # Sample a number of random images for manual quality check
+        # The number is set in config.yml as quality_check_sample_size
+        n_images_to_sample = self.sampling_parameters["quality_check_sample_size"]
+        
+        logger.info(f'Sampling {n_images_to_sample} images for quality check.. \n')
+        
+        quality_check_images = SmartSampling.get_n_random_images_per_date(grouped_images_by_date, n_images_to_sample)
+        
+        logger.info(f'Quality check images: {quality_check_images} \n')
+
+        for key, values in quality_check_images.items():
+            for value in values:
+                copy_file(
+                    "/" + key + "/" + value, str(self.input_structured_folder), str(self.customer_quality_check_folder)
+                )
+        
+    def sample_images_for_retraining(self, grouped_images_by_date: Dict[str, List[str]]) -> None:
+        """
+        Samples images for retraining purposes from the structured input folder. The function first collects images 
+        from the database that are above a specified confidence score threshold. It then categorizes these images into 
+        bins based on detection counts and samples a specific ratio of images from each bin for every date. The sampled 
+        images are then copied to the customer retraining folder.
+
+        Parameters
+        ----------
+        grouped_images_by_date : Dict[str, List[str]]
+            A dictionary mapping dates (in 'YYYY-MM-DD_HH_MM_SS' string format) to lists of image filenames. 
+            These filenames are used to query the database and collect detection data for the corresponding images.
+            
+        Returns
+        -------
+        None
+        """
+        
+        # Collect images above the confidence score threshold from the database
```

Comment:
Comment is redundant due to good naming on function and variables.
I'm a big fan (even if some people don't agree) of a no comments between the code, my theory is that if you need to write a comment then it means that that code needs to be refactored on a better structure.